src/lib/reasoners/satml_frontend.ml

```diff
@@ -1309,8 +1310,12 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
       Inst.add_terms env.inst
         (E.max_ground_terms_rec_of_form gf.E.ff) gf;
     try
-      assert (SAT.decision_level env.satml == 0);
-      let _updated = assume_aux ~dec_lvl:0 env [gf] in
+      let _updated =
+        assume_aux
+          ~dec_lvl:(SAT.decision_level env.satml)
```

Comment:
I think ideally this should be either `dec_lvl:0` or `dec_lvl:(SAT.push_level env.satml)` (the 2nd is probably better) here, but alas SatML would not support this with `--no-minimal-bj`.